runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContext.java

```diff
@@ -313,19 +325,24 @@ public Closeable enterState(ExecutionState newState) {
       if (isDataflowProcessElementState) {
         DataflowExecutionState newDFState = (DataflowExecutionState) newState;
         if (newDFState.getStepName() != null && newDFState.getStepName().userName() != null) {
-          if (this.activeMessageMetadata != null) {
-            recordActiveMessageInProcessingTimesMap();
+          synchronized (this) {
+            if (this.activeMessageMetadata != null) {
```

Comment:
```suggestion
```

nit: We can remove this check here since `recordActiveMessageInProcessingTimesMap` also has a null check inside